qa/L0_trt_reformat_free/trt_reformat_free_test.py

```diff
@@ -0,0 +1,271 @@
+            self.assertTrue(
+                np.array_equal(output1_np, expected_output1_np),
+                "OUTPUT0 expected: {}, got {}".format(expected_output1_np,
+                                                      output1_np))
+
+    def test_nobatch_chw32_input(self):
+        model_name = "plan_nobatch_CHW32_LINEAR_float32_float32_float32"
+        input_np = np.arange(26, dtype=np.float32).reshape((13, 2, 1))
+        expected_output0_np = input_np + input_np
+        expected_output1_np = input_np - input_np
+        # FIXME Can't generate reformat-free model of this specification
+        reformatted_input_np = input_np  #reformat("CHW32", input_np)
+
+        # Use shared memory to bypass the shape check in client library, because
+        # for non-linear format tensor, the data buffer is padded and thus the
+        # data byte size may not match what is calculated from tensor shape
+        inputs = []
+        inputs.append(tritonhttpclient.InferInput('INPUT0', [13, 2, 1], "FP32"))
+        self.add_reformat_free_data_as_shared_memory("input0", inputs[-1],
+                                                     reformatted_input_np)
+        inputs.append(tritonhttpclient.InferInput('INPUT1', [13, 2, 1], "FP32"))
+        self.add_reformat_free_data_as_shared_memory("input1", inputs[-1],
+                                                     reformatted_input_np)
+
+        outputs = []
+        outputs.append(
+            tritonhttpclient.InferRequestedOutput('OUTPUT0', binary_data=True))
+        outputs.append(
+            tritonhttpclient.InferRequestedOutput('OUTPUT1', binary_data=True))
+
+        results = self.triton_client.infer(model_name=model_name,
+                                           inputs=inputs,
+                                           outputs=outputs)
+        # Validate the results by comparing with precomputed values.
+        output0_np = results.as_numpy('OUTPUT0')
+        output1_np = results.as_numpy('OUTPUT1')
+        self.assertTrue(
+            np.array_equal(output0_np, expected_output0_np),
+            "OUTPUT0 expected: {}, got {}".format(expected_output0_np,
+                                                  output0_np))
+        self.assertTrue(
+            np.array_equal(output1_np, expected_output1_np),
+            "OUTPUT0 expected: {}, got {}".format(expected_output1_np,
+                                                  output1_np))
+
+    def test_chw32_input(self):
+        model_name = "plan_CHW32_LINEAR_float32_float32_float32"
+        for bs in [1, 8]:
+            input_np = np.arange(26 * bs, dtype=np.float32).reshape(
+                (bs, 13, 2, 1))
+            expected_output0_np = input_np + input_np
+            expected_output1_np = input_np - input_np
+            reformatted_input_np = reformat("CHW32", input_np)
+
+            # Use shared memory to bypass the shape check in client library,
```

Comment:
Do you think this warrants another API to set tensor data that skips data byte sizes checks? We must not force users to use shared memory if they want to use reformat-free models. We can do this as another PR.

Reply:
Yes, I think that will be good. We can discuss about how to achieve it, in addition to it, perhaps we should extend the format field in model config to include the TRT formats, so the user may inquire the input format and preprocess the data accordingly if needed. Currently we assume that the user knows what the model is expecting beforehand.

Reply:
DLIS-1662